ci/unit_tests/workspace_addjson/main_test.py

```diff
@@ -0,0 +1,32 @@
+
+import os, unittest, sys, codecs, json
+
+import workspace_addjson
+
+class TestMain(unittest.TestCase):
+
+    ''' Tests if particular file is found by getFilesAtPath function '''
+    def test_main(self):
+
+        cmd = 'cp ./tests/test_data/workspace.json ./tests/test_data/workspace_forAddJsonTest.json'
+        os.system(cmd)
+
+        workspaceFile = 'workspace_forAddJsonTest.json'
+        directory = './tests/test_data/'
+        jsonAddPath = './tests/test_data/jsonToAdd.json'
+        targetNode = 'parent'
+        referenceResult = './tests/test_data/workspace_result.json'
+
+        workspace_addjson.main(['-w', workspaceFile, '-d', directory , '-j', jsonAddPath, '-t', targetNode, '-v'])
+
+        # compare the new result with the reference result
+        with codecs.open(os.path.join(directory, workspaceFile), 'r', encoding='utf8') as inputpath:
+            newWorkspace = json.load(inputpath)
+
+        with codecs.open(os.path.join(referenceResult), 'r', encoding='utf8') as inputpath:
+            referenceWorkspace = json.load(inputpath)
+
+        referenceWorkspaceString = json.dumps(referenceWorkspace)
+        newWorkspaceString = json.dumps(newWorkspace)
+
+        assert referenceWorkspaceString == newWorkspaceString
```

Comment:
Will it work if `targetNode` will be item in list? What if no `targetNode` is found? -> more tests needed

Reply:
if target node won't be find, the results json won't change

Reply:
Ok, the first cannot happen and if no `targetNode` found it prints a warning.